salt/config/__init__.py

```diff
@@ -2560,8 +2560,6 @@ def is_profile_configured(opts, provider, profile_name):
 
     if driver not in non_image_drivers:
         required_keys.append('image')
-    elif driver == 'vmware':
-        required_keys.append('clonefrom')
```

Comment:
How about this for now so we can get going with the commit?:

- I rename guest_id to image.
- I revert to clonefrom being a required key.

It merges the creation support but makes it unavailable for now. Then we can create a new issue with a discussion on how to best map the VMware driver to be consistent with other cloud drivers.